### PR TITLE
 Add support for user mode XRC-SRQ's

### PR DESCRIPTION
**Pull request by mkalderon**
rdma-core part of the user mode XRC-SRQ's support
Kernel patch: https://patchwork.kernel.org/patch/11678337/

Signed-off-by: Yuval Basson <ybason@marvell.com>
Signed-off-by: Michal Kalderon <mkalderon@marvell.com>

### CODE REVIEW

**Comment by mkalderon** [comment]
@jgunthorpe @rleon 
Kernel side of this was applied: https://lore.kernel.org/linux-rdma/20200722102339.30104-1-ybason@marvell.com/
thanks,
Michal

File: providers/qedr/qelr_hsi_rdma.h [hunk 1]
```diff
@@ -163,6 +168,7 @@ enum rdma_cqe_type
 	RDMA_CQE_TYPE_REQUESTER,
 	RDMA_CQE_TYPE_RESPONDER_RQ,
 	RDMA_CQE_TYPE_RESPONDER_SRQ,
+	RDMA_CQE_TYPE_RESPONDER_XRC_SRQ,
```
Comment (rleon) [inline providers/qedr/qelr_hsi_rdma.h:171]: Are you sure that addition here won't break anything?
You moved RDMA_CQE_TYPE_INVALID from the previous value.

Comment (mkalderon) [reply providers/qedr/qelr_hsi_rdma.h:171]: @rleon yes, this enum is hsi with FW. Before support for XRC_SRQ rdma-core treated a cqe_type
of this type as invalid, now with support it is a valid CQE. thanks

Comment (rleon) [reply providers/qedr/qelr_hsi_rdma.h:171]: Last question, won't you have a compatibility problem with old FW and new rdma-core?

Thanks

Comment (mkalderon) [reply providers/qedr/qelr_hsi_rdma.h:171]: @rleon good point. I just checked with FW team and apparently CQE_TYPE_INVALID is never used and was never used. So also with old-FW there is no flow that can reach a cqe with this type. thanks!

**Comment by rleon** [comment]
Thanks

**Pull request merged**